### ISSUE TITLE
OSError cannot allocate memory during report generation

### ISSUE THREAD

**Issue by hillaryj** [cisagov/cyhy-reports#64]
# 🐛 Summary #

During report generation, a report failed due to `OSError: [Errno 12] Cannot allocate memory`. This happened on one of the longest-generation-time reports, which may have contributed.

## To reproduce ##

1. Run report generation
2. See if one or more reports fail to generate due to an `OSError`

## Expected behavior ##

We probably want to catch this error and retry that report's generation. This is not a code error but instead is an environment limitation, so we'll want to handle it for future runs.

## Any helpful log output or screenshots ##

Paste the results here:

```console
Traceback (most recent call last):
  File "/usr/local/bin/cyhy-report", line 11, in <module>
    generate_report.main()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 3393, in main
    was_encrypted, results = generator.generate_report()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 341, in generate_report
    self.__generate_final_pdf()
  File "/usr/local/lib/python2.7/dist-packages/cyhy_report/customer/generate_report.py", line 3299, in __generate_final_pdf
    ["xelatex", "report.tex"], stdout=output, stderr=subprocess.STDOUT
  File "/usr/lib/python2.7/subprocess.py", line 168, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 916, in _execute_child
    self.pid = os.fork()
OSError: [Errno 12] Cannot allocate memory
```

**Comment by dav3r** [comment]
Since we are not really investing time into the current version of CyHy, I think the simplest solution here is to simply reduce the number of concurrent reporting threads (see #63) from 24 back down to 16.

**Comment by dav3r** [comment]
This issue has been resolved to our satisfaction via #119.
